homeassistant/components/wallbox/sensor.py

```diff
@@ -145,6 +146,10 @@ class WallboxSensorEntityDescription(SensorEntityDescription):
         device_class=SensorDeviceClass.CURRENT,
         state_class=SensorStateClass.MEASUREMENT,
     ),
+    CHARGER_OCPP_STATUS_KEY: WallboxSensorEntityDescription(
+        key=CHARGER_OCPP_STATUS_KEY,
+        translation_key=CHARGER_OCPP_STATUS_KEY
+    ),
```

Comment:
What does this sensor return?

Reply:
Can we transform that integer into a human readable string?

Reply:
Sadly I have not managed to find any information to enumerate the states, I assume this is a similar problem with the Current mode sensor as that is not enumerated too. I only know that 2 seems to be when OCPP is configured and not connected and 4 is when the connection is established ( and appears in the portal as "Connection accepted" ). I would be guessing at any other states.

Reply:
but returning "connection_accepted" is more user friendly than returning "4". With translations we can then translate the state.

You can make this an enum sensor and we should do our due dilligence trying to figure out what the states mean and put that in the `options` field of the sensor. This will also enable users to easily automate on that since we already know the possible options.

```py
STATES = {
    2: "not_connected",
    4: "connection_accepted",
}
```

and then in the sensor you can do `STATES.get(state)` and if it's not found in that list it will return `None` which will render the sensor as `Unknown`.

So with this approach, when the sensor returns anything else than 2 or 4, it would render Unknown. If someone then finds an alternative state and wants to read it out, they can create an issue.

I'd rather have a very user friendly sensor that is able to show what it can do, than having a sensor that returns something that is vague for the user.

Reply:
@iPeel @joostlek 

export enum OCPP_CONNECTION_STATUS {
  DISCONNECTED = 0,
  NOT_CONFIGURED = 1,
  CONNECTION_REJECTED = 2,
  CONNECTION_ERROR = 3,
  CONNECTED = 4
}
 from the source of the wallbox portal

Reply:
Sounds like perfect data to make this an enum sensor :)

Reply:
I think CONNECTION_REJECTED is a misnomer and it doesn't tally with my obervations. Following the rabbit hole:

`export const ocppStatuses = () => [
  {
    code: OCPP.DISCONNECTED,
    label: i18n.global.t('mywb.common.disabled'),
    ids: [OCPP_CONNECTION_STATUS.DISCONNECTED, OCPP_CONNECTION_STATUS.NOT_CONFIGURED]
  },
  {
    code: OCPP.CONNECTING,
    label: i18n.global.t('mywb.common.connecting'),
    ids: [
      OCPP_CONNECTION_STATUS.CONNECTION_REJECTED
    ]
  },
  {
    code: OCPP.ERROR,
    label: i18n.global.t('mywb.ocpp.connection-rejected'),
    ids: [OCPP_CONNECTION_STATUS.CONNECTION_ERROR]
  },
  {
    code: OCPP.CONNECTED,
    label: i18n.global.t('mywb.ocpp.connection-accepted'),
    ids: [OCPP_CONNECTION_STATUS.CONNECTED]
  }
]`

So:

`
OCPP_STATES = {

  0: "disabled",
  1: "disconnected",
  2: "connecting",
  3: "connection_error",
  4: "connected"

}
`
It would be useful to inspect what a wallbox with unconfigured OCPP returns, 0 = disabled is just a logical guess.

Reply:
I don't have OCPP configured, mine returns a 1 for OCPP status.